modules/plottinggl/src/processors/orthographicaxis2d.cpp

```diff
@@ -89,14 +97,34 @@ OrthographicAxis2D::OrthographicAxis2D()
 }
 
 void OrthographicAxis2D::process() {
-    utilgl::activateTargetAndClearOrCopySource(outport_, inport_, ImageType::ColorDepth);
-
     const auto dims = outport_.getDimensions();
-    const size2_t lowerLeft(margins_.getLeft(), margins_.getBottom());
-    const size2_t upperRight(dims.x - 1 - margins_.getRight(), dims.y - 1 - margins_.getTop());
 
     const auto padding = axisMargin_.get();
 
+    const utilgl::ClearColor clearColor{backgroundColor_.get()};
+    utilgl::activateAndClearTarget(outport_);
+    if (auto source = inport_.getData()) {
+        const auto scissors =
```

Comment:
The scissor test is only enabled for this if statement and not the subsequent rendering. Is this intended? Maybe put in a comment that you want to clip the input image/layer to the plot area/margins.
Also, the scissor test only applies to rendered fragments, so in case `copyRepresentationsTo()` uses a pixel transfer or directly copies the textures it wont have an effect.

Reply:
fixed